ait/core/server/server.py

```diff
@@ -0,0 +1,255 @@
+
+        self._load_streams()
+        self._load_plugins()
+
+        self.broker.inbound_streams = self.inbound_streams
+        self.broker.outbound_streams = self.outbound_streams
+        self.broker.servers = self.servers
+        self.broker.plugins = self.plugins
+
+        # defining greenlets that need to be joined over
+        self.greenlets = ([self.broker] +
+                           self.broker.plugins +
+                           self.broker.inbound_streams +
+                           self.broker.outbound_streams)
+
+    def wait(self):
+        """
+        Starts all greenlets for concurrent processing.
+        Joins over all greenlets that are not servers.
+        """
+        for greenlet in (self.greenlets + self.servers):
+            log.info("Starting {} greenlet...".format(greenlet))
+            greenlet.start()
+
+        gevent.joinall(self.greenlets)
+
+    def _load_streams(self):
+        """
+        Reads, parses and creates streams specified in config.yaml.
+        """
+        common_err_msg = 'No valid {} telemetry stream configurations found. '
+        specific_err_msg = {'inbound': 'No telemetry will be received (or displayed).',
+                            'outbound': 'No telemetry will be published.'}
+        err_msgs = {}
+
+        for stream_type in ['inbound', 'outbound']:
+            err_msgs[stream_type] = common_err_msg.format(stream_type) + specific_err_msg[stream_type]
+            streams = ait.config.get('server.{}-streams'.format(stream_type))
+
+            if streams is None:
+                log.warn(err_msgs[stream_type])
+            else:
+                for index, s in enumerate(streams):
+                    try:
+                        strm = self._create_stream(s['stream'], stream_type)
+                        if stream_type == 'inbound' and type(strm) == PortInputStream:
+                            self.servers.append(strm)
+                        elif stream_type == 'inbound':
+                            self.inbound_streams.append(strm)
+                        elif stream_type == 'outbound':
+                            self.outbound_streams.append(strm)
+                        log.info('Added {} stream {}'.format(stream_type, strm))
+                    except Exception:
+                        exc_type, value, tb = sys.exc_info()
+                        log.error('{} creating {} stream {}: {}'.format(exc_type,
+                                                                        stream_type,
+                                                                        index,
+                                                                        value))
+        if not self.inbound_streams and not self.servers:
+            log.warn(err_msgs['inbound'])
+
+        if not self.outbound_streams:
+            log.warn(err_msgs['outbound'])
+
+    def _create_stream(self, config, stream_type):
+        """
+        Creates a stream from its config.
```

Comment:
Let's make sure we get all the function docstrings updated to this format. Although a little more whitespace between the categories of stuff and a more elaborate description of attributes would also be nice =)